app/middleware/renderer.js

```diff
@@ -1,7 +1,51 @@
+const log = require('../lib/logger');
+const messages = require('../lib/messages');
+
 function findHelp(req, res) {
   res.render('find-help');
 }
 
+function location(req, res) {
+  if (res.locals.locationHeading === null) {
+    return res.render('error');
+  }
+  return res.render('location');
+}
+
+function results(req, res) {
+  res.render('results');
+}
+
+function emptyPostcodePage(req, res) {
+  log.debug('Empty Search');
+  res.locals.errorMessage = messages.emptyPostcodeMessage();
+  location(req, res);
+}
+
+function postcodeError(error, postcode, res, next) {
+  log.debug({ postcode }, 'Error in postcode');
+  res.locals.errorMessage = messages.technicalProblems();
+  next(error);
+}
+
+function invalidPostcodePage(postcode, req, res) {
```

Comment:
Can the parameter name be `location` rather than `postcode`? It makes more sense when viewed with the log statement which is `Location failed...`.
Also, for consistency with the other functions it would be good have the parameter list begin with `req, res, ...`.